libraries/browser-tracker-core/src/helpers/browser_props.ts

```diff
@@ -1,8 +1,11 @@
+/* Separator used for dimension values e.g. widthxheight */
+const DIMENSION_SEPARATOR = 'x';
+
 export function getBrowserProperties() {
   return {
-    viewport: detectViewport(),
-    documentSize: detectDocumentSize(),
-    resolution: screen.width + 'x' + screen.height,
+    viewport: floorDimensionFields(detectViewport()),
+    documentSize: floorDimensionFields(detectDocumentSize()),
+    resolution: floorDimensionFields(detectScreenResolution()),
```

Comment:
Have we confirmed when the fractional values actually occur (which useragent/circumstances)? It would be good to have that concrete before we throw away data... and before someone inevitably asks why it's wrong for that random user agent.

Reply:
> throw away data
In what case will this occur ? Or you mean, not allowing the fractional part to get through ?

The change in behaviour from this PR is only that fractional values will be rounded down to the closest int for these attributes.

Reply:
Yes, just that we're discarding that fractional info. Its existence means _something_ (bot?), I just don't know what, and we're throwing that data point away by rounding it. :stuck_out_tongue:

Reply:
Thought about it a bit, let me provide some of my thoughts:
- If the `innerWidth` or any other value is fractional probably means there is some user-agent bug or unexpected behaviour. It does not automatically signal bot traffic though.
- Does a condition like a fractional value on width/height make it a bad **event** that needs to be manually reviewed _without having a clear response if it is indeed bad_ ? That's where I am stuck in the logic.
- We cannot change the type of the parameter to accept fractional values as it is part of the atomic table and will probably be a breaking change.
- What we can is introduce a `browser_context` model change _(not sure we even have it modeled yet)_ where we allow for fractional values to be pushed in and analyzed normally without ending in the bad queue.

WDYT ?

Reply:
To be clear, I have no problem with this change and agree with it - we just need to know when/why the fractional value occurs if possible, so it's explainable why the captured value is different from the real value.

Like if atomic.events had a maxLength on useragent of like 120 and we made the trackers .slice(0,120) the value so it was always good, we would probably want to document some of the known cases of it being over 120 characters to at least understand why the values are partial sometimes, right?

Reply:
Yeap, I will add a task for that so it will remain on our radar! I totally agree